File: sc/src/main/scala/Solution_283.scala
class Solution_283 {
  def moveZeroes(nums: Array[Int]): Unit = {
    var cursor = 0
    for (i <- nums.indices) {
      if (nums(i) != 0) {
        nums(cursor) = nums(i)
        cursor += 1
      }
    }
    for {
      j <- cursor until nums.length
    } nums(j) = 0
//    println(nums.map(_.toInt).toList)
  }
}

File: kt/src/main/kotlin/id/algo/Solution_283.kt
package id.algo

class Solution_283 {
    fun moveZeroes(nums: IntArray): Unit {
        var cursor = 0
        for(i in nums.indices) {
            if(nums[i] != 0) {
                nums[cursor++] = nums[i]
            }
        }
        for (j in cursor until nums.size) {
            nums[j] = 0
        }
    }
}
